.github/workflows/check.yml

```diff
@@ -12,4 +12,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - run: ./gradlew checkstyleMain checkstyleTest checkWithCodenarc runUnitTests
+      - name: Set up the JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
```

Comment:
Why are we not using latest java LTS version, java-21 ?

Reply:
Actually we are using Java 8 everywhere :) Java 17 is used only to run gradle process. I prefer not to rush in here and use latest LTS, because gradle adds java support with some delay. We always can update with next gradle update